### DISCUSSION TITLE
absent properties sets to zero for primitives by default

### DISCUSSION THREAD

**Discussion by Boeing1337** [FasterXML/jackson-databind#4586] [General]
### Describe your Issue

When we use constructor injections
`record DTORecord(int id) {}` 
 with empty body request `{}` it should be consider as bad request, property id is mandatory. 
 What is the reason to consider absent value as zero? Zero is a value, why does framework decides for me which value I send?
 Am I missing some logic?
 so why FAIL_ON_MISSING_CREATOR_PROPERTIES is not a default behavior?

**Comment by cowtowncoder** [comment]
If you want it to be required, use `@JsonProperty(required=true)` or enable `FAIL_ON_MISSING_CREATOR_PROPERTIES`.

As to why default of `0`: this is the default value for `int`s (if you leave a POJO `int` field unassigned it'll have that value) -- since something must be passed, Jackson uses what is generally considered the default (since `null` won't work for primitives).
For historical reasons properties in Jackson are considered optional.
If your preference is different, use configuration.

**Reply by Boeing1337** [reply]
Thank you for answer.
